test/built-ins/TypedArrays/from/mapfn-arguments.js

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+id: sec-%typedarray%.from
+description: >
+  Assert mapfn arguments
+info: >
+  22.2.2.1 %TypedArray%.from ( source [ , mapfn [ , thisArg ] ] )
+
+  ...
+  10. Repeat, while k < len
+    ...
+    c. If mapping is true, then
+      i. Let mappedValue be ? Call(mapfn, T, « kValue, k »).
+  ...
+includes: [testTypedArray.js]
+---*/
+
+var source = [42, 43, 44];
+
+testWithTypedArrayConstructors(function(TA) {
+  var results = [];
+  var mapfn = function(kValue, k) {
+    results[k] = kValue;
```

Comment:
This test is a little imprecise for a few reasons:

1. By using the second argument as an array key, we are unable to distinguish between string and number values. The implementation could be invoking the callback with, e.g., `"1"` instead of `1`, and this test would still pass.
2. We cannot determine the order that function calls were made
3. There is no assertion of the number of arguments

This could be addressed by storing the values of each parameter and the `arguments` length as elements of three independent arrays.